pkg/sql/catalog/tabledesc/constraint.go

```diff
@@ -430,6 +430,10 @@ func newConstraintCache(
 			} else {
 				c.all = append(c.all, fk)
 				c.fks = append(c.fks, fk)
+				if m.WriteAndDeleteOnly() {
+					c.allEnforced = append(c.allEnforced, fk)
+					c.fksEnforced = append(c.fksEnforced, fk)
+				}
```

Comment:
You are right and I tried that, but it breaks two existing unit tests ([test1](https://github.com/cockroachdb/cockroach/blob/0682851cda2cd29c7e9e033ec7e1c4d804ef5d06/pkg/sql/schema_changer_test.go#L5673) and [test2](https://github.com/cockroachdb/cockroach/blob/0682851cda2cd29c7e9e033ec7e1c4d804ef5d06/pkg/sql/schema_changer_test.go#L5769)) for legacy schema changer. Legacy schema changer has different semantics: the constraint does not move to being enforced until after we've backfilled the columns. It's an implicit state that we don't model (or, we model differently in the new schema changer). 
If we added the same thing for check constraints when building the cache, the inserts will fail because the check mutation is enforced while the unit test assumes that the insert should succeed (bc it's before the column is backfilled). 

I thought about deleting those two unit tests altogether but I think they are getting at something important so I didn't.

Reply:
Seems legit. Please add commentary to explain this.

Reply:
already did!